src/DataCache.js

```diff
@@ -228,6 +193,23 @@ class DataCache {
         this.logger.trace(`The data cache has been cleared.`);
         this.count = 0;
         this.data = {};
+        this.loaded.clear();
+    }
+
+    /**
+     * Record that the given file name has already been loaded.
+     */
+    fileLoaded(fileName) {
```

Comment:
IMHO, 
What about `loadFile` or `addFile` ?  actually adding is happen at the moment.
I was confused between fileLoaded and wasLoaded  at the first time.
just my thought.

Reply:
Okay, I'll make better method names